manifests/charts/ai-gateway-helm/crds/aigateway.envoyproxy.io_llmroutes.yaml

```diff
@@ -0,0 +1,149 @@
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: |-
+          LLMRoute combines multiple LLMBackends and attaching them to Gateway(s) resources.
+
+          This serves as a way to define a "unified" LLM API for a Gateway which allows downstream
+          clients to use a single schema API to interact with multiple LLM backends.
+
+          The InputSchema is used to determine the structure of the requests that the Gateway will
+          receive. And then the Gateway will route the traffic to the appropriate LLMBackend based
+          on the output schema of the LLMBackend while doing the other necessary jobs like
+          upstream authentication, rate limit, etc.
+        properties:
+          apiVersion:
+            description: |-
+              APIVersion defines the versioned schema of this representation of an object.
+              Servers should convert recognized schemas to the latest internal value, and
+              may reject unrecognized values.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+            type: string
+          kind:
+            description: |-
+              Kind is a string value representing the REST resource this object represents.
+              Servers may infer this from the endpoint the client submits requests to.
+              Cannot be updated.
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: Spec defines the details of the LLM policy.
+            properties:
+              backendRefs:
+                description: |-
+                  BackendRefs lists the LLMBackends that this LLMRoute will route traffic to.
+                  The namespace is "local", i.e. the same namespace as the LLMRoute.
+                items:
+                  description: LLMBackendLocalRef is a reference to a LLMBackend resource
+                    in the "local" namespace.
+                  properties:
+                    name:
+                      description: Name is the name of the LLMBackend in the same
+                        namespace as the LLMRoute.
+                      type: string
+                  required:
+                  - name
+                  type: object
+                maxItems: 128
+                type: array
+              inputSchema:
+                description: |-
+                  APISchema specifies the API schema of the input that the target Gateway(s) will receive.
+                  Based on this schema, the ai-gateway will perform the necessary transformation to the
+                  output schema specified in the selected LLMBackend during the routing process.
+
+                  Currently, the only supported schema is OpenAI as the input schema.
```

Comment:
would it be less problematic to remove this text constraint and introduce bedrock later once supported?

Reply:
the constraint will be enforced in the cel validation rule that happens at k8s API server - will do the follow up soon

Reply:
good comment anyways adrian as usual